# **Trabalho II - Inferência Estatística**
# **Responsável: Luma Chen**

# **1- Média amostral e variância amostral**

In [ ]:
%%R
mu<-2
sigma<-2
n<-10
asim<-10
xbar<-rep(NA, asim)
xvar<-rep(NA, asim)
for (i in 1:asim)
{
  print(i)
  set.seed(i)
  x<-rnorm(n, mu, sigma)
  xbar[i]<-mean(x)
  print(xbar[i])
  xvar[i]<-var(x)
  print(xvar[i])
}

[1] 1
[1] 2.264406
[1] 2.437258
[1] 2
[1] 2.422303
[1] 3.880803
[1] 3
[1] 1.865729
[1] 2.997949
[1] 4
[1] 3.133059
[1] 4.387793
[1] 5
[1] 1.842297
[1] 3.627733
[1] 6
[1] 2.210716
[1] 3.374709
[1] 7
[1] 2.207951
[1] 6.417745
[1] 8
[1] 1.101362
[1] 4.621303
[1] 9
[1] 1.561963
[1] 1.815755
[1] 10
[1] 1.018686
[1] 1.959404


(i) A média amostral é um estimador não viesado da média populacional, pois o valor esperado dela é sempre igual ao da média populacional. Além disso, as médias calculadas no código são sempre próximas da média populacional, mu= 2, variando por conta do desvio padrão.

(ii) Sim, a variância amostral é um estimador
não enviesado para a variância populacional. Porque o resultado esperado da variância amostral é igual ao da variância populacional. No caso, o resultado do código acima apresentou valores próximos da variância populacional, que é 4, ou seja, sendo 4 o resultado esperado de cada variância amostral.

In [ ]:
%%R
mu<-2
sigma<-2
n<-10
asim<-100
xbar<-rep(NA, asim)
xvar<-rep(NA, asim)
for (i in 1:asim)
{
  print(i)
  set.seed(i)
  x<-rnorm(n, mu, sigma)
  xbar[i]<-mean(x)
  print(xbar[i])
  xvar[i]<-var(x)
  print(xvar[i])
}

[1] 1
[1] 2.264406
[1] 2.437258
[1] 2
[1] 2.422303
[1] 3.880803
[1] 3
[1] 1.865729
[1] 2.997949
[1] 4
[1] 3.133059
[1] 4.387793
[1] 5
[1] 1.842297
[1] 3.627733
[1] 6
[1] 2.210716
[1] 3.374709
[1] 7
[1] 2.207951
[1] 6.417745
[1] 8
[1] 1.101362
[1] 4.621303
[1] 9
[1] 1.561963
[1] 1.815755
[1] 10
[1] 1.018686
[1] 1.959404
[1] 11
[1] 1.539875
[1] 4.174121
[1] 12
[1] 1.065572
[1] 4.005256
[1] 13
[1] 3.200108
[1] 1.943137
[1] 14
[1] 3.508412
[1] 3.708359
[1] 15
[1] 2.357318
[1] 3.643844
[1] 16
[1] 2.267633
[1] 3.236919
[1] 17
[1] 2.354561
[1] 2.738028
[1] 18
[1] 2.714518
[1] 7.02458
[1] 19
[1] 1.734601
[1] 1.838908
[1] 20
[1] 1.275338
[1] 6.887702
[1] 21
[1] 2.395333
[1] 6.016097
[1] 22
[1] 2.958946
[1] 3.954774
[1] 23
[1] 3.386323
[1] 2.352382
[1] 24
[1] 2.014443
[1] 1.344915
[1] 25
[1] 1.651246
[1] 3.550987
[1] 26
[1] 2.103497
[1] 3.543264
[1] 27
[1] 2.713726
[1] 6.052825
[1] 28
[1] 1.484561
[1] 5.300914
[1] 29
[1] 1.925851
[1] 5.579744
[1] 30
[1] 1.676637
[1] 4.507405
[1] 31
[1] 2.559551


(iii) O resultado obtido com amostras de tamanhos maiores, provou o estimador como sendo não enviesado, pois os valores da média e variância amostrais, ao final, se aproximaram bem mais dos seus respectivos valores esperados.

In [ ]:
import numpy as np
import random
from statistics import mean
from numpy import var
import scipy.stats as stats

mu=2
sigma=2
n=10
asim = 10
xbar=np.repeat(np.nan, asim)
xvar=np.repeat(np.nan, asim)
for i in range(1, asim + 1):
  print(i)
  random.seed(i)
  x=np.random.normal(mu, sigma, n)
  xbar[i- 1]=mean(x)
  print("Média:", xbar[i-1])
  xvar[i-1]=var(x)
  print("Variância", xvar[i-1])

1
Média: 2.354502870583631
Variância 7.666108364607797
2
Média: 2.7233879359512216
Variância 4.900254682854903
3
Média: 2.6842495311762304
Variância 2.1885805969721126
4
Média: 1.7802092608446838
Variância 6.889917615295995
5
Média: 0.780355081732066
Variância 3.3008268929091633
6
Média: 1.9323576327067673
Variância 5.29036697965234
7
Média: 1.2658208599057634
Variância 4.138697466463422
8
Média: 0.6900972679449441
Variância 4.660645657946579
9
Média: 1.970779545043257
Variância 5.084588668362859
10
Média: 1.434666965938027
Variância 3.0955771588292995


In [ ]:
import numpy as np
import random
from statistics import mean
from numpy import var
import scipy.stats as stats

mu= 2
sigma=2
n= 10
asim = 100
xbar=np.repeat(np.nan, asim)
xvar=np.repeat(np.nan, asim)
for i in range(1, asim + 1):
  print(i)
  random.seed(i)
  x=np.random.normal(mu, sigma, n)
  xbar[i- 1]=mean(x)
  print("Média:", xbar[i-1])
  xvar[i-1]=var(x)
  print("Variância", xvar[i-1])
  media= mean(xvar)
  print("Média da variância amostral: ", media)

1
Média: 2.2447614000544553
Variância 2.353370581085207
Média da variância amostral:  nan
2
Média: 1.144583621405984
Variância 1.4549152765903965
Média da variância amostral:  nan
3
Média: 1.8246634088701696
Variância 3.986798285729555
Média da variância amostral:  nan
4
Média: 2.6372820406905775
Variância 7.105825023092396
Média da variância amostral:  nan
5
Média: 2.786756401922251
Variância 0.9255887911842551
Média da variância amostral:  nan
6
Média: 1.57471854138011
Variância 5.700662956530014
Média da variância amostral:  nan
7
Média: 2.377171326001583
Variância 4.059849425638386
Média da variância amostral:  nan
8
Média: 1.7917008038819715
Variância 3.617508882288361
Média da variância amostral:  nan
9
Média: 2.0090563999080078
Variância 3.150091665512968
Média da variância amostral:  nan
10
Média: 1.8308148950404783
Variância 4.132672274769535
Média da variância amostral:  nan
11
Média: 2.5011910116918243
Variância 8.606587237429885
Média da variância amostral:  nan
12
Média: 0

(v) Conforme o tamanho das amostras aumentam, os valores obtidos da média de todas as variâncias amostrais de cada amostra, se aproxima mais do valor esperado, que é sigma ao quadrado.

In [ ]:
%%R
mu<-2
sigma<-2
n<-10
asim<-1000
xbar<-rep(NA, asim)
xbar2<-rep(NA, asim)
xbar3<-rep(NA, asim)
xvar<-rep(NA, asim)
xvar2<-rep(NA, asim)
xvar3<-rep(NA, asim)
for (i in 1:asim)
{
  print(i)
  set.seed(i)
  x<-rnorm(n, mu, sigma)
  x2<-rnorm(n*10, mu, sigma)
  x3<-rnorm(n*100, mu, sigma) 
  xbar[i]<-mean(x)
  xbar2[i]<-mean(x2)
  xbar3[i]<-mean(x3)
  xvar[i]<-var(x)
  xvar2[i]<-var(x2)
  xvar3[i]<-var(x3)
}
print("Média 1:")
print(mean(xbar))
print("Média 2:")
print(mean(xbar2))
print("Média 3:")
print(mean(xbar3))
print("Variância 1:")
print(var(xbar))
print("Variância 2:")
print(var(xbar2))
print("Variância 3:")
print(var(xbar3))
print("Média da variância 1:")
print(mean(xvar))
print("Média da variância 2:")
print(mean(xvar2))
print("Média da variância 3:")
print(mean(xvar3))


[1] 1
[1] 2
[1] 3
[1] 4
[1] 5
[1] 6
[1] 7
[1] 8
[1] 9
[1] 10
[1] 11
[1] 12
[1] 13
[1] 14
[1] 15
[1] 16
[1] 17
[1] 18
[1] 19
[1] 20
[1] 21
[1] 22
[1] 23
[1] 24
[1] 25
[1] 26
[1] 27
[1] 28
[1] 29
[1] 30
[1] 31
[1] 32
[1] 33
[1] 34
[1] 35
[1] 36
[1] 37
[1] 38
[1] 39
[1] 40
[1] 41
[1] 42
[1] 43
[1] 44
[1] 45
[1] 46
[1] 47
[1] 48
[1] 49
[1] 50
[1] 51
[1] 52
[1] 53
[1] 54
[1] 55
[1] 56
[1] 57
[1] 58
[1] 59
[1] 60
[1] 61
[1] 62
[1] 63
[1] 64
[1] 65
[1] 66
[1] 67
[1] 68
[1] 69
[1] 70
[1] 71
[1] 72
[1] 73
[1] 74
[1] 75
[1] 76
[1] 77
[1] 78
[1] 79
[1] 80
[1] 81
[1] 82
[1] 83
[1] 84
[1] 85
[1] 86
[1] 87
[1] 88
[1] 89
[1] 90
[1] 91
[1] 92
[1] 93
[1] 94
[1] 95
[1] 96
[1] 97
[1] 98
[1] 99
[1] 100
[1] 101
[1] 102
[1] 103
[1] 104
[1] 105
[1] 106
[1] 107
[1] 108
[1] 109
[1] 110
[1] 111
[1] 112
[1] 113
[1] 114
[1] 115
[1] 116
[1] 117
[1] 118
[1] 119
[1] 120
[1] 121
[1] 122
[1] 123
[1] 124
[1] 125
[1] 126
[1] 127
[1] 128
[1] 129
[1] 130
[1] 131
[1] 132
[1] 133
[1] 134
[1] 135
[1] 136
[1] 137
[1] 138
[1] 

(vi) Após analisar as 3 amostras de tamanhos diferentes com suas respectivas, médias, variâncias e média das variâncias, pude conlcuir que quanto maior o tamanho da amostra, mais próximo o valor consegue chegar do resultado esperado, média sendo 2 e média das variâncias sendo 4. Além disso, quanto maior a amostra, mais a variância diminuia, ou seja, prova que os valores de cada amostra ficavam mais próximos e menos discrepantes entre si.

In [ ]:
mu=2
sigma=2
n=10
asim = 1000
xbar=np.repeat(np.nan, asim)
xbar2=np.repeat(np.nan, asim)
xbar3=np.repeat(np.nan, asim)
xvar=np.repeat(np.nan, asim)
xvar2=np.repeat(np.nan, asim)
xvar3=np.repeat(np.nan, asim)
for i in range(1, asim + 1):
  print(i)
  random.seed(i)
  x=np.random.normal(mu, sigma, n)
  x2=np.random.normal(mu, sigma, n*10)
  x3=np.random.normal(mu, sigma, n*100)
  xbar[i- 1]=mean(x)
  xbar2[i- 1]=mean(x2)
  xbar3[i- 1]=mean(x3)
  xvar[i-1]=var(x)
  xvar2[i-1]=var(x2)
  xvar3[i-1]=var(x3)

print("Média 1: ", mean(xbar))
print("Média 2: ", mean(xbar2))
print("Média 3: ", mean(xbar3))
print("Variância 1: ", var(xbar))
print("Variância 2: ", var(xbar2))
print("Variância 3: ", var(xbar3))
print("Média da variância 1: ", mean(xvar))
print("Média da variância 2: ", mean(xvar2))
print("Média da variância 3: ", mean(xvar3))

1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128
129
130
131
132
133
134
135
136
137
138
139
140
141
142
143
144
145
146
147
148
149
150
151
152
153
154
155
156
157
158
159
160
161
162
163
164
165
166
167
168
169
170
171
172
173
174
175
176
177
178
179
180
181
182
183
184
185
186
187
188
189
190
191
192
193
194
195
196
197
198
199
200
201
202
203
204
205
206
207
208
209
210
211
212
213
214
215
216
217
218
219
220
221
222
223
224
225
226
227
228
229
230
231
232
233
234
235
236
237
238
239
240
241
242
243
244
245
246
247
248
249
250
251
252
253
254
255
256
257
258
259
260
261
262
263
264
265
266
267
268
269
270
271
272
273
274
275
276
277


# **2- Intervalo de confiança para μ**

In [8]:
%%R
library (MASS)
head (survey)
print(head)
help(survey)
read.csv('survey.csv')
alturaAlunos = survey$Height

#z-score
media<- mean(alturaAlunos, na.rm=TRUE)
sigma<- sd(alturaAlunos, na.rm=TRUE)
tamanho = length(na.omit(alturaAlunos)) 
erro = sigma/sqrt(tamanho)
conf = 0.95
y = (1 - conf)/2
qnorm = qnorm(y)
intervaloInf = media - 1.96*erro
intervaloSup= media + 1.96*erro
print("Média:")
print(media)
print("Intervalo de confiança do z-score:")
print(intervaloInf)
print(intervaloSup)

#t-score
t.test(na.omit(alturaAlunos), alternative = c("two.sided"), conf.level = 0.95) 

function (x, ...) 
UseMethod("head")
<bytecode: 0x558e030620b0>
<environment: namespace:utils>
[1] "Média:"
[1] 172.3809
[1] "Intervalo de confiança do z-score:"
[1] 171.0458
[1] 173.7159

	One Sample t-test

data:  na.omit(alturaAlunos)
t = 253.07, df = 208, p-value < 2.2e-16
alternative hypothesis: true mean is not equal to 0
95 percent confidence interval:
 171.0380 173.7237
sample estimates:
mean of x 
 172.3809 



Sendo o tamanho da amostra maior que 30, o intervalo de confiança mais próximo da média (172,3809) foi o z-score, uma vez que ele funciona melhor para amostras grandes. A diferença foi sutil, mas perceptível.

# **3- Intervalo de confiança para** p

In [44]:
import numpy as np
import pandas as pd
import math

df_survey = pd.read_csv('survey.csv', sep="[;,]", engine='python')
saved_column = df["W.Hnd"]
numCan= sum(1 for i in saved_column if i == 'Left')
numDes= sum(1 for i in saved_column if i == 'Right')
total= numCan + numDes
mediaCan= numCan/total
mediaDes= numDes/total
intervalo= 1.96 * math.sqrt((mediaCan * mediaDes)/ total)
print("O intervalo de confiança está entre:", mediaCan - intervalo, "e", mediaCan + intervalo)

O intervalo de confiança está entre: 0.04240607054219365 e 0.1101363023391623


Pelo menos 10 alunos são destros e pelo menos 10 são canhotos. Ao todo, temos 18 alunos canhotos e com o intervalo de confiança, no nível de 95%, aproximadamente entre: 4,24% e 11,01%. Ou seja, pudemos notar que a minoria dos alunos é canhota, com uma porcentagem total não ultrapassando os 11% dos alunos da universidade.